### DISCUSSION TITLE
General Questions

### DISCUSSION THREAD

**Discussion by andrew4328** [Lambda-Mountain-Compiler-Backend/lambda-mountain#629] [Q&A]
LM is growing as a project and there are some learning curves to overcome. All questions are welcome here on this thread.

**Comment by alex-s168** [comment]
why is there HashtableEq and HashtableIs

**Comment by andrew4328** [comment]
Eq uses == for equality testing. Is uses is for equality testing. The `{}` syntax uses HashtableEq.

The AST -> Type index uses HashtableIs because that is faster and also semantically we don't want structural equality.

**Comment by andrew4328** [comment]
I would like to parameterize this eventually, so the type of hashtable would be `Hashtable<k,v,equality-comparison=$"==",hash-function=$"deep-hash">`

or something like that. Maybe a type class idk.

**Comment by andrew4328** [comment]
Update: this has since been collapsed into `Hashtable<Type,Type>` for deep equality vs `Hashtable<Sized<Type>,Type>` for shallow equality.
